DPB_converger.py

```diff
@@ -0,0 +1,363 @@
+    :return: List of matching file paths
+    """
+    matching_files = []
+    for path, subdirs, files in os.walk(directory):
+        for name in files:
+            if target in name.lower():
+                if exclusion_string not in name.lower():
+                    matching_files.append(os.path.join(path, name))
+    return matching_files
+
+
+def define_swft_command(file_path, tau, exp_displacement, exp_noise_rate, max_displacement, max_displacement_pp,
+                        p_bleach, p_switch, precision, diff_limit):
+    """swft command per file"""
+    file_command = 'swft.exe ' + file_path + ' --tau ' + '"' + tau + '" ' + '--out_values "all" '
+    file_command += '--exp_displacement ' + '"' + exp_displacement + '"' + ' '
+    file_command += '--exp_noise_rate ' + '"' + str(exp_noise_rate) + '"' + ' '
+    file_command += '--max_displacement ' + '"' + max_displacement + '"' + ' '
+    file_command += '--max_displacement_pp ' + '"' + max_displacement_pp + '"' + ' '
+    file_command += '--p_bleach ' + '"' + p_bleach + '"' + ' '
+    file_command += '--p_switch ' + '"' + p_switch + '"' + ' '
+    file_command += '--precision ' + '"' + str(precision) + '"' + ' '
+    file_command += '--diffraction_limit ' + '"' + diff_limit + '"' + ' '
+    return file_command
+
+
+def gather_Data(filepaths):
+    """
+    Collects the displacement and pBleach data of all notebook outputs in the given list of directories
+    """
+    #gathers a list of all relevant files
+    displacement_files = []
+    bleach_files = []
+    for working_directory in filepaths:
+        displacement_files += get_matching_files(working_directory, 'displacement', 'histogram')
+        bleach_files += get_matching_files(working_directory, 'p_bleach', 'histogram')
+    displacement_data = []
+    bleach_data = []
+    # reads the values and puts them into one list, then returns it
+    for file in displacement_files:
+        disp = float(open(file, 'r').readlines()[1].split('	')[0])
+        displacement_data.append(disp)
+    for file in bleach_files:
+        bleach = float(open(file, 'r').readlines()[1].split('	')[0])
+        bleach_data.append(bleach)
+    return displacement_data, bleach_data
+
+
+def write_swift(config_path, exp_displacement, p_bleach):
+    """
+    Effectively does what the main method of the write_swift_batch.py does,
+    however the parameters exp_displacement and p_bleach are no longer read from file but given as arguments
+    """
+    file_paths = []
+    n_file_paths = []
+    precision_vals = []
+    exp_noise_rate_vals = []
+    config = configparser.ConfigParser()
+    config.sections()
+    config.read(config_path)
+    """
+    The script expects the files precision and exp_noise_rate to be in the folder
+    \PreAnalysisParameter\parameters, unlike the write_swift_batch.py which allows for a directory to be specified.
+    However this is where BatchPreAnalysis places them.
```

Comment:
Let's add flexibility. If file paths are defined they are used, if they are defined as None, the default path chosen by batchPreAnalysis is used. 

```
[PRECISION_INDIVIDUAL]
precision01 = None
precision02 = None
```